src/shared/version_op.c

```diff
@@ -661,11 +661,28 @@ os_info *get_unix_version()
             } else if(strcmp(strtok_r(buff, "\n", &save_ptr),"Darwin") == 0){
                 info->os_platform = strdup("darwin");
 
-                if (cmd_output_ver = popen("sw_vers -productName", "r"), cmd_output_ver) {
-                    if(fgets(buff, sizeof(buff) - 1, cmd_output_ver) == NULL){
-                        mdebug1("Cannot read from command output (sw_vers -productName).");
-                    } else {
-                        w_strdup(strtok_r(buff, "\n", &save_ptr), info->os_name);
+                if (cmd_output_ver = popen("system_profiler SPSoftwareDataType", "r"), cmd_output_ver) {
+                    while (fgets(buff, sizeof(buff), cmd_output_ver) != NULL) {
+                        char *key = strtok_r(buff, ":", &save_ptr);
+                        if (key) {
+                            const char *expected_key = "SystemVersion";
```

Comment:
This should be `System Version` right?
```suggestion
                            const char *expected_key = "System Version";
```

Reply:
This ins't necessary bad... I need to remove the left spaces so I'm ussing a `strip` function which removes all spaces, including the middle space.

Now I'm chaging this to `w_strtrim` function, which doesn't removes the middle space.